src/main/kotlin/io/perfometer/runner/ThreadPoolScenarioRunner.kt

```diff
@@ -1,36 +1,46 @@
 package io.perfometer.runner
 
-import io.perfometer.dsl.ScenarioBuilder
-import io.perfometer.http.*
+import io.perfometer.dsl.PauseStep
+import io.perfometer.dsl.RequestStep
+import io.perfometer.dsl.HttpStep
 import io.perfometer.http.client.HttpClient
-import io.perfometer.statistics.*
-import io.perfometer.statistics.printer.StatisticsPrinter
+import io.perfometer.statistics.ConcurrentQueueScenarioStatistics
+import io.perfometer.statistics.PauseStatistics
+import io.perfometer.statistics.RequestStatistics
+import io.perfometer.statistics.ScenarioStatistics
 import java.time.Duration
 import java.time.Instant
-import java.util.concurrent.*
+import java.util.concurrent.CompletableFuture
+import java.util.concurrent.ExecutorService
+import java.util.concurrent.Executors
+import java.util.concurrent.TimeUnit
 
-internal class DefaultScenarioRunner(
+internal class ThreadPoolScenarioRunner(
         private val httpClient: HttpClient,
-        private val statisticsPrinter: StatisticsPrinter,
+        private val scenarioStatistics: ScenarioStatistics = ConcurrentQueueScenarioStatistics(Instant.now())
 ) : ScenarioRunner {
-    private lateinit var scenarioStatistics: ScenarioStatistics
 
-    override fun run(scenario: ScenarioBuilder, configuration: RunnerConfiguration) {
-        scenarioStatistics = ConcurrentQueueScenarioStatistics(Instant.now())
-        runScenario(scenario, configuration)
-    }
-
-    private fun runScenario(scenario: ScenarioBuilder, configuration: RunnerConfiguration) {
-        val scenarioExecutor = Executors.newFixedThreadPool(configuration.threads)
+    override fun runUsers(userCount: Int, block: () -> Unit) {
```

Comment:
Consider  changing the name of the paramter, maybe `action` or `task` will be a bit more descriptive?

Reply:
done